src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -532,10 +532,19 @@ private static bool CanBindToTheseConstructorParameters(ParameterInfo[] construc
             Type keyType = dictionaryType.GenericTypeArguments[0];
             Type valueType = dictionaryType.GenericTypeArguments[1];
             bool keyTypeIsEnum = keyType.IsEnum;
-
-            if (keyType != typeof(string) && !keyTypeIsEnum)
-            {
-                // We only support string and enum keys
+            bool keyTypeIsIntegralNonPtr =
```

Comment:
```suggestion
            bool keyTypeIsInteger =
```

I don't think we need to say "non Ptr" in the variable name. A comment should suffice.

Reply:
Thoughts on this one?

Reply:
Just pushed.